.github/workflows/e2e.yml

```diff
@@ -285,163 +282,35 @@ jobs:
-            perl -i -pe's/newArchEnabled=false/newArchEnabled=true/g' gradle.properties
-            echo 'New Architecture enabled'
-          fi
-          [[ "${{ matrix.build-type }}" == "production" ]] && CONFIG='Release' || CONFIG='Debug'
-          echo "Building $CONFIG"
-          ./gradlew ":app:assemble$CONFIG" -PreactNativeArchitectures=x86
-
-      - name: Build iOS App
-        if: ${{ matrix.platform == 'ios' }}
-        working-directory: test/react-native/versions/${{ matrix.rn-version }}/RnDiffApp/ios
-        run: |
-          [[ "${{ matrix.build-type }}" == "production" ]] && CONFIG='Release' || CONFIG='Debug'
-          echo "Building $CONFIG"
-          mkdir -p "DerivedData"
-          derivedData="$(cd "DerivedData" ; pwd -P)"
-          set -o pipefail && xcodebuild \
-            -workspace RnDiffApp.xcworkspace \
-            -configuration "$CONFIG" \
-            -scheme RnDiffApp \
-            -destination 'platform=iOS Simulator,OS=${{ matrix.runtime }},name=${{ matrix.device }}' \
-            ONLY_ACTIVE_ARCH=yes \
-            -derivedDataPath "$derivedData" \
-            build \
-            | tee xcodebuild.log \
-            | xcbeautify --quieter --is-ci --disable-colored-output
-
-      - name: Archive Android APK
-        if: matrix.platform == 'android' && matrix.build-type == 'production'
-        run: |
-          BUILD_PATH=test/react-native/versions/${{ matrix.rn-version }}/RnDiffApp/android/app/build/outputs/apk/release
-          BUILD_NAME=app-release.apk
-          tar -cvf apk.tar -C $BUILD_PATH $BUILD_NAME
-
-      - name: Archive iOS APP
-        if: matrix.platform == 'ios' && matrix.build-type == 'production'
-        run: |
-          BUILD_PATH=test/react-native/versions/${{ matrix.rn-version }}/RnDiffApp/ios/DerivedData/Build/Products/Release-iphonesimulator
-          BUILD_NAME=RnDiffApp.app
-          tar -cvf app.tar -C $BUILD_PATH $BUILD_NAME
+      - name: Build Plain RN ${{ matrix.rn-version }} App
+        run: ./scripts/e2e.mjs ${{ matrix.platform }} --build
 
-      - name: Upload App APK
-        if: matrix.platform == 'android' && matrix.build-type == 'production'
-        uses: actions/upload-artifact@v4
-        with:
-          name: ${{ matrix.rn-version }}-${{ matrix.rn-architecture }}-${{ matrix.engine }}-${{ matrix.platform }}-${{ matrix.build-type }}-${{ matrix.ios-use-frameworks }}-app-package
-          path: apk.tar
-          retention-days: 1
-
-      - name: Upload App APP
-        if: matrix.platform == 'ios' && matrix.build-type == 'production'
+      - name: Upload App
+        if: matrix.build-type == 'production'
         uses: actions/upload-artifact@v4
         with:
           name: ${{ matrix.rn-version }}-${{ matrix.rn-architecture }}-${{ matrix.engine }}-${{ matrix.platform }}-${{ matrix.build-type }}-${{ matrix.ios-use-frameworks }}-app-package
-          path: app.tar
+          path: test/e2e/RnDiffApp.ap*
```

Comment:
`.app` on macOS is a dir and the upload action doesn't upload dirs, or at least it did not in the past.

Reply:
dirs work since upload-artifact@v2 IIIRC

Reply:
Ahh, okay, thanks, then this is fine.